File: src/main/java/com/cabservice/service/AuthService.java
package com.cabservice.service;

import com.cabservice.dto.request.LoginRequestDTO;
import com.cabservice.dto.request.UserRequestDTO;
import com.cabservice.dto.response.LoginResponseDTO;
import com.cabservice.dto.response.UserResponseDTO;

public interface AuthService {

    void startRegistration(UserRequestDTO userRequestDTO);

    void verifyRegistrationEmail(
            String email,
            String otp
    );

    UserResponseDTO verifyRegistrationMobile(
            String mobileNo,
            String otp
    );

    LoginResponseDTO loginUser(LoginRequestDTO request);

    void sendForgotPasswordOtp(String email);

    void verifyForgotPasswordOtp(
            String email,
            String otp
    );

    void resetPassword(
            String email,
            String newPassword
    );
}

File: ui/src/services/authService.js
import api from "./api";

export const startRegistration = async (data) => {
	const response = await api.post(
		"/auth/register",
		data
	);

	return response.data;
};

export const verifyEmailOtp = async (
	email,
	otp
) => {
	const response = await api.post(
		"/auth/register/verify-email",
		null,
		{
			params: {
				email,
				otp,
			},
		}
	);

	return response.data;
};

export const verifyMobileOtp = async (
	mobileNo,
	otp
) => {
	const response = await api.post(
		"/auth/register/verify-mobile",
		null,
		{
			params: {
				mobileNo,
				otp,
			},
		}
	);

	return response.data;
};

export const login = async (data) => {
	const response = await api.post(
		"/auth/login",
		data
	);

	return response.data;
};

export const sendForgotPasswordOtp = async (email) => {

	const response = await api.post(
		"/auth/forgot-password/send-otp",
		{
			email
		}
	);

	return response.data;
};

export const verifyForgotPasswordOtp = async (
	email,
	otp
) => {

	const response = await api.post(
		"/auth/forgot-password/verify-otp",
		{
			email,
			otp
		}
	);

	return response.data;
};

export const resetPassword = async (
	email,
	newPassword
) => {

	const response = await api.post(
		"/auth/forgot-password/reset",
		{
			email,
			newPassword
		}
	);

	return response.data;
};